internal/manifest/exec_context.go

```diff
@@ -0,0 +1,90 @@
+package manifest
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.autokitteh.dev/autokitteh/internal/resolver"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type execContext struct {
+	log      Log
+	client   sdkservices.Services
+	resolver resolver.Resolver
+
+	projects     map[string]sdktypes.ProjectID
+	integrations map[string]sdktypes.IntegrationID
+	envs         map[string]sdktypes.EnvID
+	connections  map[string]sdktypes.ConnectionID
+}
+
+func (c *execContext) resolveProjectID(ctx context.Context, name string) (sdktypes.ProjectID, error) {
+	if pid, ok := c.projects[name]; ok {
+		return pid, nil
+	}
+
+	sdkName, err := sdktypes.ParseName(name)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := c.client.Projects().GetByName(ctx, sdkName)
+	if err != nil {
+		return nil, err
+	}
+
+	pid := sdktypes.GetProjectID(p)
+
+	c.projects[name] = pid
+
+	return pid, nil
```

Comment:
Feel free to keep this, but IMO when number of empty lines == number of code lines, you're using too many whitespaces.  :-P

Reply:
"fixed"